binding.gyp

```diff
@@ -1,5 +1,13 @@
 # This file inherits default targets for Node addons, see https://github.com/nodejs/node-gyp/blob/master/addon.gypi
 {
+  # https://github.com/springmeyer/gyp/blob/master/test/make_global_settings/wrapper/wrapper.gyp
+  'make_global_settings': [
+    ['CXX', '<(module_root_dir)/mason_packages/.link/bin/clang++'],
+    ['CC', '<(module_root_dir)/mason_packages/.link/bin/clang'],
+    ['LINK', '<(module_root_dir)/mason_packages/.link/bin/clang++'],
+    ['AR', '<(module_root_dir)/mason_packages/.link/bin/llvm-ar'],
+    ['NM', '<(module_root_dir)/mason_packages/.link/bin/llvm-nm']
+  ],
```

Comment:
@GretaCB this is a big change. It moves us to setting the defaults for the compiler toolchain here rather than depending on environment variables only. The motivation was to try to reduce the use of bash as much as possible. This combined with `mason-js` now allows us to use a consistent compiler without needing to use `./scripts/setup.sh` on travis.

Reply:
Perhaps we can add something in the docs about how to swap out compilers if someone wanted to use something else and not pull down automatically from mason.

Reply:
Also, did a basic google search but couldnt find anything specific. What exactly does `make_global_settings` do? Assuming something like sets `CXX` etc as global env vars that reference these bin commands?